tendrl/node_agent/manager/manager.py

```diff
@@ -19,22 +82,114 @@ def __init__(self):
+        self._discovery_thread = TopLevelEvents(self)
+        self.persister = Persister()
 
     def stop(self):
         LOG.info("%s stopping" % self.__class__.__name__)
         self._user_request_thread.stop()
+        self._discovery_thread.stop()
+        self.persister.stop()
 
     def start(self):
         LOG.info("%s starting" % self.__class__.__name__)
         self._user_request_thread.start()
+        self._discovery_thread.start()
+        self.persister.start()
 
     def join(self):
         LOG.info("%s joining" % self.__class__.__name__)
         self._user_request_thread.join()
+        self._discovery_thread.join()
+        self.persister.join()
+
+    def on_pull(self, raw_data):
+        LOG.info("on_pull, Updating node metadata data")
+        self.persister.update_node_metadata(
+            NodeMetadata(
+                updated=str(time.time()),
+                node_machine_uuid=raw_data["node_machine_uuid"],
+                node_uuid=raw_data["node_uuid"],
+                fqdn=raw_data["os"]["FQDN"],
+            )
+        )
+        if "os" in raw_data:
+            LOG.info("on_pull, Updating OS data")
+            node = raw_data['os']
+            self.persister.update_os(
+                Os(
+                    updated=str(time.time()),
+                    os=node["Name"],
+                    os_version=node["OSVersion"],
+                    kernel_version=node["KernelVersion"],
+                    selinux_mode=node["SELinuxMode"],
+                    node_uuid=raw_data["node_uuid"],
+                )
+            )
+        if "memory" in raw_data:
+            LOG.info("on_pull, Updating memory")
+            memory = raw_data['memory']
+            self.persister.update_memory(
+                Memory(
+                    updated=str(time.time()),
+                    total_size=memory["TotalSize"],
+                    total_swap=memory["SwapTotal"],
+                    node_uuid=raw_data["node_uuid"],
+                )
+            )
+
+        if "cpu" in raw_data:
+            LOG.info("on_pull, Updating cpu")
+            cpu = raw_data['cpu']
+            self.persister.update_cpu(
+                Cpu(
+                    updated=str(time.time()),
+                    model=cpu["Model"],
+                    vendor_id=cpu["VendorId"],
+                    model_name=cpu["ModelName"],
+                    architecture=cpu["Architecture"],
+                    cores_per_socket=cpu["CoresPerSocket"],
+                    cpu_op_mode=cpu["CpuOpMode"],
+                    cpu_family=cpu["CPUFamily"],
+                    cpu_count=cpu["CPUs"],
+                    node_uuid=raw_data["node_uuid"],
+                )
+            )
+
+
+def configure_tendrl_uuid():
+    # check if valid uuid is already present in tendrl_node_id_file.
+    # if not present generate one and update the file
+    try:
+        with open(TENDRL_NODE_ID_FILE) as f:
```

Comment:
timestamp required while updating, also delete the older node_agent_key_$oldtimestamp

at any given point in time, there should be only one /etc/tendrl/node_agent_key_$latesttimestamp